src/nupic/types/Sdr.hpp

```diff
@@ -506,6 +506,23 @@ class SparseDistributedRepresentation : public Serializable
      */
     void load(std::istream &inStream) override;
 
+
+    // New serialization library **Cereal**.
+    // Do NOT call this externally!  Instead use methods save/load.
+    template<class Archive>
+    void save_ar(Archive & archive) const
+    {
+        archive( dimensions, getSparse() );
+    }
+
+    template<class Archive>
+    void load_ar(Archive & archive)
+    {
+        archive( dimensions_, sparse_ );
+        initialize( dimensions_ );
+        setSparseInplace();
```

Comment:
this serialization does look nice!

Reply:
This is what I would expect most of them to look like.  Very simple.